samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -1002,8 +1157,11 @@ def _map_s3_sources_to_functions(s3_hash_to_source: Dict[str, str], cfn_resource
         CloudFormation resources
     """
     for _, resource in cfn_resources.items():
-        if resource.get("Type") == CFN_AWS_LAMBDA_FUNCTION:
-            code = resource.get("Properties").get("Code")
+        resource_type = resource.get("Type")
```

Comment:
can you add unit test cases for `_map_s3_source_to_functions` for the lambda layer use case. like this one https://github.com/aws/aws-sam-cli/blob/8c8bc2e54ba1a03b5bc6d0a4cf7bb8d67ea8d3e3/tests/unit/hook_packages/terraform/test_prepare_hook.py#L882